src/middlelayer/token_request_types.rs

```diff
@@ -16,3 +23,56 @@ impl GetToken {
         Ok(DieselUlid::from_str(&self.0.token_id)?)
     }
 }
+
+impl CreateToken {
+    pub fn get_resource(&self) -> Result<Option<ObjectMapping<DieselUlid>>> {
+        Ok(match &self.0.permission {
+            Some(perm) => match &perm.resource_id {
+                Some(res) => match res {
+                    ResourceId::ProjectId(id) => {
+                        Some(ObjectMapping::PROJECT(DieselUlid::from_str(id)?))
+                    }
+                    ResourceId::CollectionId(id) => {
+                        Some(ObjectMapping::COLLECTION(DieselUlid::from_str(id)?))
+                    }
+                    ResourceId::DatasetId(id) => {
+                        Some(ObjectMapping::DATASET(DieselUlid::from_str(id)?))
+                    }
+
+                    ResourceId::ObjectId(id) => {
+                        Some(ObjectMapping::OBJECT(DieselUlid::from_str(id)?))
+                    }
+                },
+                None => None,
+            },
+            None => return Err(anyhow!("No permissions provided")),
+        })
+    }
+
+    pub fn get_expiry(&self) -> Result<NaiveDateTime> {
+        let exp = match &self.0.expires_at {
+            Some(exp) => NaiveDateTime::from_timestamp_opt(exp.seconds, exp.nanos.try_into()?)
+                .ok_or_else(|| anyhow!("Timestamp conversion error"))?,
+            None => Utc::now()
+                .naive_local()
+                .checked_add_months(Months::new(6))
```

Comment:
In the case of unspecified expiry time, "unlimited" normally applies. So far this has been 10 years.